Release

Michael-Reefe opened the pull request.

Michael-Reefe committed:
Added telescope & dome modules
Edits from demo

Michael-Reefe committed:
Added toggle for Slave dome to scope function

Michael-Reefe committed:
Shortened name of function

Michael-Reefe committed:
Reader for config file w/ parameters

Michael-Reefe committed:
No changes, just adding it back in

Michael-Reefe committed:
Removed accidental mention of focuser

Michael-Reefe committed:
Added more config parameters

Michael-Reefe committed:
Added more readiness checks.  Cleaned up jog.

Michael-Reefe committed:
Edits for threading.  Still buggy.

Michael-Reefe committed:
Small change.  Still buggy.

Michael-Reefe committed:
Shawn's practice example

Michael-Reefe committed:
Removed erroneous threading implementation.
To be added back soon.

Michael-Reefe committed:
Now supports different timezones

Michael-Reefe committed:
Removed bad threading; fixed timezones

Michael-Reefe committed:
Now supports restarting from a crash without overwriting old images

Michael-Reefe committed:
Adding back important gitignore files

Michael-Reefe committed:
Added astropy converstion to/from J2000

Michael-Reefe committed:
Now with camera threading!

Michael-Reefe committed:
Added threading.
Dome & Telescope now in observation_run.

Michael-Reefe committed:
Added support for more ra/dec formats in tickets

Michael-Reefe committed:
Minor edits; better cooler adjusting

Michael-Reefe committed:
Minor logging changes to debug threading

Michael-Reefe committed:
Threading changes.  Still some issues.

Michael-Reefe committed:
Removed accidental comment out for shutter

Michael-Reefe committed:
Removed accidental comment out for shutter

Michael-Reefe committed:
UNTESTED: COM object now dispatched within their own threads,
eliminating pythoncom

Michael-Reefe committed:
Threading nearly finalized

Michael-Reefe committed:
Minor fix

Michael-Reefe committed:
Added weather checker

Michael-Reefe committed:
Small edit

Michael-Reefe committed:
Minor time change for cooler settle

Michael-Reefe committed:
Small shutter wait change

Michael-Reefe committed:
New/improved weather_checker uses radar from weather.com

Michael-Reefe committed:
Directory for storing weather files for weather checks

Michael-Reefe committed:
Committing gitignore

Michael-Reefe committed:
Improved weather check using regular expressions

Michael-Reefe committed:
Small change

Michael-Reefe committed:
Changed time for cooler settle

Michael-Reefe committed:
Bugfixes and crash checks

Kakon24 committed:
Added basic CLI, HAS NOT BEEN TESTED

Michael-Reefe committed:
Date change

Michael-Reefe committed:
Hardware class

Michael-Reefe committed:
New hardware class

Michael-Reefe committed:
New logger module

Michael-Reefe committed:
Removed hard-coded directory

Michael-Reefe committed:
Added comments and logging messages

oalfaro2 committed:
Merge branch 'feature/CLI' into michaels_branch

Merged from feature/CLI

oalfaro2 committed:
Added focuser (UNTESTED)
Made small changes to cli -- we may need to rewrite most/all imports

Michael-Reefe committed:
Changed to relative imports

Michael-Reefe committed:
CLI fixes & adjustments

Michael-Reefe committed:
Removed outdated crash guide

Michael-Reefe committed:
Small shutter status fix

Michael-Reefe committed:
More optional CLI arguments

Michael-Reefe committed:
Added extra format for ra/dec input

Michael-Reefe committed:
Small change to make weather slightly less sensitive

Michael-Reefe committed:
Now accepts multiple tickets

Michael-Reefe committed:
Fixed multi-ticket inputs--added sanity check

Michael-Reefe committed:
Owen added IRAFStarFind to the focuser

Michael-Reefe committed:
Now accepts folder input for tickets as well.  Shutdown prompt w/
timeout of 1 minute.

Michael-Reefe committed:
Fixed weather check so it exits when code is done

Michael-Reefe committed:
Focuser now with filereader capabilities, implemented

Michael-Reefe committed:
Small fixes & edits.  Focuser still dicey.

Michael-Reefe committed:
Small bugfix

Michael-Reefe committed:
Small fixes

Michael-Reefe committed:
Focuser FWHM fixes

Michael-Reefe committed:
Comments

Michael-Reefe committed:
Small fixes to fwhm and cooler.  Focuser now working, but algorithm may
need adjusting.
[Also not sure how accurate the fwhm's from MaxIm DL are...]

Michael-Reefe committed:
Fixed thread stopping error

Michael-Reefe committed:
Uncommented accidental comment

Michael-Reefe committed:
Disconnect fix

Michael-Reefe committed:
Focusing changes

Michael-Reefe committed:
Reworked focusing.
Now there is the Focuser class, which is NOT subclassed from hardware but
is instead utilized by the FocusProcedures class, which is subclassed
from hardware.  We only need FocusProcedures for observation_run.

Michael-Reefe committed:
Small changes; untested

Michael-Reefe committed:
Shutdown is now a CLI argument

Michael-Reefe committed:
Everything ok test: Checks weather, Sun elevation, and hardware
connections

Michael-Reefe committed:
Reworked focuser and focusprocedures.
Now they are 2 separate threads, both inherited from hardware.

Michael-Reefe committed:
Small focus fixes & new MaxIm DL crash handler

Michael-Reefe committed:
Replaced os.system with subprocess.call

Michael-Reefe committed:
Better MaxIm DL crash handler; preliminary guider steps

Michael-Reefe committed:
Guider additions & work on sunset calculations (still unfinished)

Michael-Reefe committed:
Sunrise/sunset pause and resume implemented.
NOTE: If user sets noshutdown CLI parameter, it also won't shutdown
after detecting bad weather.  Working on this fix next.

Michael-Reefe committed:
Weather shutdown adjustments

Michael-Reefe committed:
Full weather & sun checker built in together.  Now if a weather/sun
check is set, the code will close up, pause, and resume when appropriate
(partially untested).

Michael-Reefe committed:
Small guider changes.  Still needs work.

oalfaro2 committed:
Added function to get FWHMs from fits file
FWHMs returned are in whole numbers

Michael-Reefe committed:
Implemented manual fwhm calculations into focuser

Michael-Reefe committed:
Owen & I made edits to fwhm and star centroid find
But they still need work

Michael-Reefe committed:
Quick bugfixes found from test run last night.
Image names, and some focuser stuff

Michael-Reefe committed:
Crash check edits and fixes

Michael-Reefe committed:
Crash checks now implemented for all hardware
Camera, telescope, dome, and focuser

Michael-Reefe committed:
Star finding, fwhm adjustments, guider update

Co-authored-by: Michael-Reefe <michael.reefe8@gmail.com>
Co-authored-by: oalfaro2 <owen.alfaro@gmail.com>

Michael-Reefe committed:
Saturation now an input from config file

Michael-Reefe committed:
Tiny fix for guider

Michael-Reefe committed:
Small weather & time fixes

Co-authored-by: Michael-Reefe <michael.reefe8@gmail.com>
Co-authored-by: oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Guider fixes & added subframe around guide star

Co-authored-by: Michael-Reefe <michael.reefe8@gmail.com>
Co-authored-by: oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Fixed subframe for the guider

Co-authored-by: Michael-Reefe <michael.reefe8@gmail.com>
Co-authored-by: oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Centroid Center of mass calculations

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Cloud coverage check WIP

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Cloud coverage checker

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Noticed some tiny errors & fixed quickly

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Noticed some tiny errors & fixed quickly

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Finalized cloud check & renamed weather to conditions

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
New Flat lamp module & darks/flats procedure

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Edited flat/dark collection -- new calibration module

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Calibration fixes & guider imiplementation

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Small fixes

Michael-Reefe committed:
Bugfixes found from testing last night

oalfaro2 committed:
New widget for easily creating observation tickets

oalfaro2 committed:
Created Observation Ticket widget (WIP)

Michael-Reefe committed:
Tiny error fix

oalfaro2 committed:
Fixed observation ticket creator widget
and added more functionality

Michael-Reefe committed:
Small adjustments

Co-authored-by: Michael-Reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Fixed guider findstar function

Co-authored-by: Michael-Reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Commented out matplotlib

Michael-Reefe committed:
Commmented out matplotlib

Michael-Reefe committed:
Added all missing documentation and comments
moved most over from owens_branch

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Focus parabola plotting and subframing

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Formatting fixes using pycharm

Michael-Reefe committed:
Small directory generalization fix

Michael-Reefe committed:
Ticket creator widget type fix
observation ticket -> observation_ticket

Michael-Reefe committed:
Added optional nocalibration CLI argument
in case you don't want to take any calibration images

Michael-Reefe committed:
Keyword argument error fix

Michael-Reefe committed:
Integer index fix for filereader utils

Michael-Reefe committed:
Filereader fix for r variable

Michael-Reefe committed:
Removed mask in favor of list syntax in filereader

Michael-Reefe committed:
Empty median data fix

Michael-Reefe committed:
Small fixes

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
More fixes for guider

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Small fixes & tests...needs more work

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Tiny fix

Michael-Reefe committed:
Focusing and guiding adjustments based on testing

Co-authored-by: Michael-Reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Small comment fix

Co-authored-by: Michael-Reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Logging rotatingfilehandler adjustment

Michael-Reefe committed:
Fixed flatlamp connection bug

Michael-Reefe committed:
More focusing & guiding adjustments
Now the guider won't pick stars if there are other stars within 100px

Michael-Reefe committed:
Reimplemented flatlamp try/except
after accidental removal

Michael-Reefe committed:
Changed matplotlib plotting for multithreading
but still getting qapplication warning...minor issue

Michael-Reefe committed:
Added backup weather.com check
for if GMU COS station is down

Michael-Reefe committed:
Slightly adjusted guiding threshold

Michael-Reefe committed:
Small changes

Michael-Reefe committed:
Small adjustments & fixes

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Guider testing w/ clocking angle (not implemented)

Michael-Reefe committed:
Adjustments to clocking angle calcs.
Still not implemented yet

Michael-Reefe committed:
Implemented clocking angle capability to guider

Michael-Reefe committed:
Minor improvements and fixes across the board

Michael-Reefe committed:
More minor fixes

Michael-Reefe committed:
Minor tweak to guider

Michael-Reefe committed:
Testing night fixes

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Testing night fixes

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Focuser and calibration fixes

Michael-Reefe committed:
More minor fixes

Michael-Reefe committed:
Release version 1.0 initial commit

Michael-Reefe committed:
Fixed telescope jog function
No more negative pulse guiding durations

Michael-Reefe committed:
More focuser fixes & additions

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Misc fixes & changes

Co-authored-by: Michael-reefe <michael.reefe8@gmail.com>
Co-authored-by: Oalfaro2 <owenalfaro@gmail.com>

Michael-Reefe committed:
Merge branch 'michaels_branch' into release

Michael-Reefe committed:
Small class changes

Michael-Reefe committed:
Merge branch 'michaels_branch' into release

Michael-Reefe committed:
Tiny, unimportant changes

Michael-Reefe committed:
Some method changes with check_connection,
and other small fixes

Michael-Reefe committed:
Added docstrings to new child check_connections
that were missing before

Michael-Reefe committed:
More small fixes

Michael-Reefe committed:
Tiny extra fix just noticed

Michael-Reefe committed:
Removed base math dependencies in favor of numpy

Michael-Reefe committed:
Argparse & other minor fixes

Michael-Reefe committed:
A few more various fixes

Michael-Reefe committed:
Focuser absolute_move docstring fix

Michael-Reefe committed:
Merge branch 'michaels_branch' into release

Michael-Reefe committed:
Updating to python 3.8 virtual environment
No noticed issues yet with the update--made some syntax improvements

Michael-Reefe committed:
Minor ticket creator widget update

Michael-Reefe committed:
Requirements & readme update

Michael-Reefe committed:
Merge branch 'michaels_branch' into release

Michael-Reefe committed:
Various fixes and removed problematic timed input
Now the only "hard-coded" directory is the output data directory in the
parameters_config file, and that can be overridden by the --data
argument of the CLI

Michael-Reefe committed:
Various other fixes

Michael-Reefe committed:
One extra tiny fix I forgot in the last commit

Michael-Reefe committed:
Merge branch 'michaels_branch' into release

Michael-Reefe committed:
Thread locks, multiple exposure times, & fixes
Also removed old test drivers

Michael-Reefe committed:
Removed temporary comments (accidentally left on)

Michael-Reefe committed:
Merge branch 'michaels_branch' into release

Michael-Reefe committed:
Various fixes from code testing nights

Michael-Reefe committed:
Merge branch 'michaels_branch' into release

Michael-Reefe committed:
Changes to make guider plots for analysis

Kakon24 commented:
Hardware class shouldn't be dependent on it's children classes.
Change to a generic .gitignore file.
Comment/delete problematic checks.

Kakon24 commented:
Delete extra feature artifacts from release branch.

Michael-Reefe committed:
Changes based on Shawn's review:
Revised gitignore to ignore .idea files & more (and itself)
Changes to hardware class to not be dependent on its children
Changes to hardware children to accomodate for the hardware changes
Moved observation ticket checks to observation_ticket.py (from
driver.py)
A few other small changes

Michael-Reefe committed:
Unimportant readme update

Michael-Reefe committed:
Merge branch 'michaels_branch' into release

Michael-Reefe committed:
Removed comments & other artifacts from my branch

Michael-Reefe committed:
A few extra final changes

Kakon24 reviewed (approved):
Great work Michael
